lib/resty/upstream/healthcheck.lua

```diff
@@ -228,6 +228,12 @@ local function check_peer(ctx, id, peer, is_backup)
             errlog("failed to connect to ", name, ": ", err)
         end
         return peer_fail(ctx, is_backup, id, peer)
+    else 
+        if ctx.type == "tcp" then
```

Comment:
I would recommend to move the `if` block a level out, so the close & return inside it is more explicitly. Like `if not ok ... end; if ctx.type == ...`.